src/reactionsystem.jl

```diff
@@ -30,19 +30,19 @@ function Catalyst.ReactionSystem(model::SBML.Model; kwargs...)  # Todo: requires
 
     algrules, obsrules, raterules = get_rules(model)
     # constant_eqs = fix_constants_at_init(model)  # Todo PL: take this out
-    # unassigned_pars = fix_unassigned_nonconstant_par_to_init(model)  # Todo PL: don't fix them to init, make them input=true
+    unassigned_pars = fix_unassigned_nonconstant_par_to_init(model)  # @Sam: if I do not do that, there are too few equations. Do you have better suggestions?
```

Comment:
What is the problem here?

Reply:
The reason why that fails is the following. Case 170 contains zero reactions, zero species and 8 parameters: S1 to S4 with `constant=false`, k1 to k4 with `constant=true`. There are rate rules for S1, S3 and S4. If I do not do `fix_unassigned_nonconstant_par_to_init`, which adds `S2(t) ~ 0.0002` to the equations, I have one equation to few (4 variables and 3 equations). [Here is a (copy-pasteable) mwe](https://gist.github.com/paulflang/716ded06decb9eb9620b2ced581d3c6b).

Atm I am adding parameters with `constant=false` to the ReactionSystem as `states`. I could add another check (namely if they participate in any Rule) before I do so. Only if they do, I would add them as `states` otherwise as `parameters`. That should fix it, right?

Reply:
Looking at that SBML there doesn't seem to be any reactions/species, so I guess this should just be a constraint system. But yes, if a variable has no equations associated with it in general you are going to have issues with most (all?) ModelingToolkit system types if you actually try to do stuff with them. So either make it a parameter in the ModelingToolkit sense, or add an equation that enforces it is constant.